packages/components/message-box/src/messageBox.ts

```diff
@@ -98,13 +98,16 @@ const showMessage = (options: any, appContext?: AppContext | null) => {
 
   for (const prop in options) {
     if (hasOwn(options, prop) && !hasOwn(vm.$props, prop)) {
-      vm[prop as string] = options[prop]
+      vm[prop as keyof ComponentPublicInstance] = options[prop]
     }
   }
 
   watch(
     () => vm.message,
-    (newVal, oldVal) => {
+    (
+      newVal: string | object | undefined,
+      oldVal: string | object | undefined
+    ) => {
```

Comment:
Probably a better way is to modify the type of message in MessageBoxState instead of modify here.